modules/core/src/main/java/org/apache/ignite/internal/util/subscription/OrderedMergePublisher.java

```diff
@@ -208,8 +208,7 @@ private void updateError(Throwable throwable) {
                 if (current == null) {
                     next = throwable;
                 } else {
-                    next = new Throwable();
-                    next.addSuppressed(current);
+                    next = current;
```

Comment:
this will not work properly. Below you have CAS which check that currest still the same and set next as a new value. But you always use the same value for next and current ones - so, the CAS from second error will be always done even it have had concurrent modification.

Reply:
Fixed, please take a look